sno/diff.py

```diff
@@ -161,13 +166,27 @@ def diff_with_writer(
         if feature_filter is not UNFILTERED:
             all_datasets = all_datasets.intersection(feature_filter.keys())
 
+        dataset_geometry_transforms = {}
+        if target_srs is not None:
+
+            dest_srs = make_srs(target_srs)
+            for dataset_path in all_datasets:
+                dataset = base_rs.get(dataset_path) or target_rs.get(dataset_path)
+                srs_wkt = dataset.srs_wkt
+                if srs_wkt is not None:
+                    src_srs = make_srs(srs_wkt)
+                    dataset_geometry_transforms[
+                        dataset.name
+                    ] = osr.CoordinateTransformation(src_srs, dest_srs)
```

Comment:
Presumably this can error?

Reply:
I haven't managed to get it to error, but looking at the GDAL code it looks like it can theoretically raise a RuntimeError. I'll wrap it with an InvalidArgument